File: src/gui/dialogs/shortcuts_dialog.py
"""
Keyboard Shortcuts Help Dialog
Shows all available keyboard shortcuts in the editor
"""

from PyQt6.QtWidgets import (
    QDialog, QVBoxLayout, QTableWidget, QTableWidgetItem, 
    QPushButton, QLabel, QHeaderView
)
from PyQt6.QtCore import Qt
from PyQt6.QtGui import QFont


class ShortcutsDialog(QDialog):
    """Dialog showing all keyboard shortcuts"""
    
    def __init__(self, parent=None):
        super().__init__(parent)
        self.setWindowTitle("Keyboard Shortcuts")
        self.setMinimumSize(500, 400)
        self._init_ui()
        
    def _init_ui(self):
        layout = QVBoxLayout(self)
        
        # Title
        title = QLabel("⌨️ Keyboard Shortcuts")
        title.setFont(QFont("Arial", 16, QFont.Weight.Bold))
        title.setAlignment(Qt.AlignmentFlag.AlignCenter)
        layout.addWidget(title)
        
        # Shortcuts table
        self.table = QTableWidget()
        self.table.setColumnCount(3)
        self.table.setHorizontalHeaderLabels(["Category", "Shortcut", "Action"])
        self.table.horizontalHeader().setSectionResizeMode(QHeaderView.ResizeMode.Stretch)
        self.table.setEditTriggers(QTableWidget.EditTrigger.NoEditTriggers)
        self.table.setAlternatingRowColors(True)
        
        # Define shortcuts
        shortcuts = [
            # Playback
            ("Playback", "Space", "Play / Pause"),
            ("Playback", "←", "Seek backward 5 seconds"),
            ("Playback", "→", "Seek forward 5 seconds"),
            
            # Sync
            ("Sync", "Space (while playing)", "Set start time for current line"),
            ("Sync", "↑", "Select previous line"),
            ("Sync", "↓", "Select next line"),
            
            # Editing
            ("Editing", "Ctrl + Z", "Undo"),
            ("Editing", "Ctrl + Y", "Redo"),
            ("Editing", "Ctrl + S", "Save project"),
            
            # Nudge
            ("Nudge", "A / D", "Nudge start time -/+ 0.1s"),
            ("Nudge", "Q / E", "Nudge end time -/+ 0.1s"),
            
            # General
            ("General", "F1", "Show this help"),
            ("General", "Escape", "Close dialog"),
        ]
        
        self.table.setRowCount(len(shortcuts))
        
        for row, (category, shortcut, action) in enumerate(shortcuts):
            self.table.setItem(row, 0, QTableWidgetItem(category))
            
            shortcut_item = QTableWidgetItem(shortcut)
            shortcut_item.setFont(QFont("Consolas", 10, QFont.Weight.Bold))
            self.table.setItem(row, 1, shortcut_item)
            
            self.table.setItem(row, 2, QTableWidgetItem(action))
        
        layout.addWidget(self.table)
        
        # Close button
        btn_close = QPushButton("Close")
        btn_close.clicked.connect(self.close)
        layout.addWidget(btn_close)
        
    def keyPressEvent(self, event):
        """Handle Escape key to close"""
        if event.key() == Qt.Key.Key_Escape:
            self.close()
        else:
            super().keyPressEvent(event)

File: cpp/src/gui/dialogs/shortcuts_dialog.cpp
/*
 * NC-KTV GUI — Shortcuts Dialog Implementation
 */

#include "shortcuts_dialog.h"

#include <QVBoxLayout>
#include <QHeaderView>
#include <QLabel>

namespace ncktv {

ShortcutsDialog::ShortcutsDialog(QWidget* parent)
    : QDialog(parent)
{
    setWindowTitle("Keyboard Shortcuts");
    setMinimumSize(480, 500);
    setModal(true);
    setupUi();
    applyTheme();
}

void ShortcutsDialog::setupUi() {
    auto* layout = new QVBoxLayout(this);
    layout->setSpacing(12);
    layout->setContentsMargins(20, 16, 20, 16);

    auto* titleLabel = new QLabel("⌨  Keyboard Shortcuts", this);
    titleLabel->setStyleSheet("font-size:18px; font-weight:bold; color:#e0e0e0; border:none;");
    layout->addWidget(titleLabel);

    m_table = new QTableWidget(this);
    m_table->setColumnCount(2);
    m_table->setHorizontalHeaderLabels({"Shortcut", "Action"});
    m_table->horizontalHeader()->setStretchLastSection(true);
    m_table->horizontalHeader()->setSectionResizeMode(0, QHeaderView::ResizeToContents);
    m_table->verticalHeader()->setVisible(false);
    m_table->setEditTriggers(QAbstractItemView::NoEditTriggers);
    m_table->setSelectionMode(QAbstractItemView::NoSelection);

    // Populate shortcuts
    QVector<QPair<QString, QString>> shortcuts = {
        {"Space",       "Set line start time (tap-to-sync)"},
        {"↑ / ↓",      "Navigate between subtitle lines"},
        {"J",           "Rewind 5 seconds"},
        {"K",           "Toggle play / pause"},
        {"L",           "Forward 5 seconds"},
        {"Ctrl+S",      "Save project"},
        {"Ctrl+Shift+S","Save project as..."},
        {"Ctrl+Z",      "Undo"},
        {"Ctrl+Y",      "Redo"},
        {"Ctrl+N",      "New project"},
        {"Ctrl+O",      "Open project"},
        {"Ctrl+E",      "Export"},
        {"Ctrl+I",      "Import subtitles"},
        {"Ctrl+B",      "Split clip at playhead"},
        {"Delete",      "Delete selected subtitle/clip"},
        {"F1",          "Show this shortcuts dialog"},
        {"Ctrl+Q",      "Quit application"},
    };

    m_table->setRowCount(shortcuts.size());
    for (int i = 0; i < shortcuts.size(); ++i) {
        auto* keyItem = new QTableWidgetItem(shortcuts[i].first);
        keyItem->setForeground(QColor("#4f46e5"));
        keyItem->setFont(QFont("Consolas", 11, QFont::Bold));
        m_table->setItem(i, 0, keyItem);
        m_table->setItem(i, 1, new QTableWidgetItem(shortcuts[i].second));
    }

    layout->addWidget(m_table, 1);

    auto* closeBtn = new QPushButton("Close", this);
    closeBtn->setFixedWidth(100);
    closeBtn->setStyleSheet("background:#4f46e5; color:white; font-weight:bold; border:none; padding:8px 16px; border-radius:4px;");
    auto* btnLayout = new QHBoxLayout();
    btnLayout->addStretch();
    btnLayout->addWidget(closeBtn);
    layout->addLayout(btnLayout);
    connect(closeBtn, &QPushButton::clicked, this, &QDialog::accept);
}

void ShortcutsDialog::applyTheme() {
    setStyleSheet(R"(
        QDialog { background:#1e1e1e; color:#d4d4d4; }
        QTableWidget { background:#2d2d30; color:#d4d4d4; gridline-color:#3f3f46; border:1px solid #3f3f46; border-radius:4px; }
        QTableWidget::item { padding:6px; }
        QHeaderView::section { background:#252526; color:#ccc; border:1px solid #3f3f46; padding:6px; }
        QPushButton { background:#333; border:1px solid #555; border-radius:4px; padding:6px 14px; color:#ccc; }
        QPushButton:hover { background:#404040; }
        QLabel { color:#ccc; border:none; }
    )");
}

} // namespace ncktv
